src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -0,0 +1,136 @@
+import java.util.stream.Stream;
+
+import static org.easymock.EasyMock.createNiceMock;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.niceMock;
+import static org.easymock.EasyMock.replay;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static uk.ac.cam.cl.dtg.segue.api.Constants.ANONYMOUS_USER;
+
+public class UsersFacadeIT extends IsaacIntegrationJupiterTest {
+    private UsersFacade usersFacade;
+
+    @BeforeAll
+    public static void beforeAll() {
+        misuseMonitor.registerHandler(RegistrationMisuseHandler.class.getSimpleName(),
+                new RegistrationMisuseHandler(emailManager, properties));
+    }
+
+    @BeforeEach
+    public void beforeEach() throws Exception {
+        misuseMonitor.resetMisuseCount("0.0.0.0", RegistrationMisuseHandler.class.getSimpleName());
+        this.usersFacade = new UsersFacade(properties, userAccountManager, logManager, userAssociationManager, misuseMonitor, userPreferenceManager, schoolListReader);
+    }
+
+    @ParameterizedTest
+    @MethodSource("validRegistrationParameters")
+    public void createUser_validRegistrationParameters(String email, String password, String dob, String familyName, String givenName) {
+        HttpSession mockSession = createNiceMock(HttpSession.class);
+        expect(mockSession.getAttribute(ANONYMOUS_USER)).andReturn(null).anyTimes();
+        expect(mockSession.getId()).andReturn("sessionIdFailure");
+        replay(mockSession);
+        HttpServletRequest mockRequest = createNiceMock(HttpServletRequest.class);
+        expect(mockRequest.getHeader("X-Forwarded-For")).andReturn("0.0.0.0").anyTimes();
+        expect(mockRequest.getSession()).andReturn(mockSession).anyTimes();
+        replay(mockRequest);
+        HttpServletResponse mockResponse = niceMock(HttpServletResponse.class);
+
+        String userObjectString = String.format("{\"registeredUser\":{\"loggedIn\":true,\"email\":\"%1$s\",\"dateOfBirth\":\"%3$s\",\"password\":\"%2$s\",\"familyName\":\"%4$s\",\"givenName\":\"%5$s\"},\"userPreferences\":{},\"passwordCurrent\":null}",
+                email, password, dob, familyName, givenName);
+
+        Response response = null;
+        try {
+            response = usersFacade.createOrUpdateUserSettings(mockRequest, mockResponse, userObjectString);
+        } catch (InvalidKeySpecException e) {
+            fail("Unhandled InvalidKeySpecException during test");
+        } catch (NoSuchAlgorithmException e) {
+            fail("Unhandled NoSuchAlgorithmException during test");
+        }
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+    }
+
+    private static Stream<Arguments> validRegistrationParameters() {
+        // Email, Password, DOB, familyName, givenName
+        return Stream.of(
+                Arguments.of("new-student@test.com", "password", "2000-01-01T00:00:00.000Z", "Test", "Test"),
```

Comment:
The only difference between the two sets of arguments is the e-mail, so that's the only thing that should be returned. The name of the function would also need changing to reflect that this is a valid email provider, rather than valid registration parameters.